modules/react/tooltip/lib/TooltipContainer.tsx

```diff
@@ -68,6 +69,28 @@ export const TooltipContainer = styled('div')<TooltipContainerProps>(
       right: 0,
       bottom: 0,
     },
+
+    // offset tooltips by 2 pixels when a keyboard focus ring is detected
+    '[data-whatinput=keyboard] &': {
+      padding: spaceNumbers.xxs + spaceNumbers.xxxs + 2,
+      ':before': {
+        margin: spaceNumbers.xxxs + 2,
+      },
+    },
+
+    // Fix offsets based on placement
+    '[data-popper-placement="top-start"] &, [data-popper-placement="bottom-start"] &': {
+      left: -spaceNumbers.xxxs,
+    },
+    '[data-popper-placement="top-end"] &, [data-popper-placement="bottom-end"] &': {
+      right: -spaceNumbers.xxxs,
+    },
+    '[data-popper-placement="left-start"] &, [data-popper-placement="right-start"] &': {
+      top: -spaceNumbers.xxxs,
+    },
+    '[data-popper-placement="left-end"] &, [data-popper-placement="right-end"] &': {
+      bottom: -spaceNumbers.xxxs,
+    },
```

Comment:
I'm not 100% sold on these changes, but here's what they do:

**Before**
![before](https://snapshots.chromatic.com/snapshots/5d854c26ba934e0020f5e98a-6138d1d3390968003a159a27/capture.png)

**After**
![after](https://snapshots.chromatic.com/snapshots/5d854c26ba934e0020f5e98a-6138f3ba390968003a1db1da/capture.png)

The subtle change is how the Tooltip aligns to the edge of the target. In the before, there's 4px spacing in the `before-start` to the left edge of the card. It is hard to see because of the border. This change removes that 4px spacing. The reason I did this was the original issue adds an additional 2px (focus ring thickness for consistency) which makes the gap more noticeable at 6px. These relative offsets fix the alignment issues. The focus ring still shifts the tooltips away from the edge, but it is much more difficult to tell. Here's what it looks like:

![focus ring](https://snapshots.chromatic.com/snapshots/5d854c26ba934e0020f5e98a-6138f3ba390968003a1db1dd/capture.png)

You can see the focus ring adds 4px on it's own because of the double ring which is done to ensure visibility against either a dark or light background. I don't think we want to align to the edge of the focus ring since that is outside the bounds of the original button.